adafruit_esp32spi/adafruit_esp32spi_wifimanager.py

```
@@ -0,0 +1,146 @@
+import adafruit_esp32spi.adafruit_esp32spi_requests as requests
+
+class ESPSPI_WiFiManager:
+    """
+    A class to help manage the Wifi connection
+    """
+    def __init__(self, esp, settings, status_neopixel=None):
+        """
+        :param ESP_SPIcontrol esp: The ESP object we are using
+        :param dict settings: The WiFi and Adafruit IO Settings (See examples)
+        :param status_neopixel: (Pptional) The neopixel pin - Usually board.NEOPIXEL (default=None)
+        :type status_neopixel: Pin
+        """
+        # Read the settings
+        self._esp = esp
+        self.debug = False
+        self.ssid = settings['ssid']
+        self.password = settings['password']
+        requests.set_interface(self._esp)
+        if status_neopixel:
+            self.neopix = neopixel.NeoPixel(status_neopixel, 1, brightness=0.2)
+        else:
+            self.neopix = None
+        self.neo_status(0)
+
+    def connect(self):
+        """
+        Attempt to connect to WiFi using the current settings
+        """
+        if self.debug:
+            if self._esp.status == adafruit_esp32spi.WL_IDLE_STATUS:
+                print("ESP32 found and in idle mode")
+            print("Firmware vers.", self._esp.firmware_version)
+            print("MAC addr:", [hex(i) for i in self._esp.MAC_address])
+            for access_pt in self._esp.scan_networks():
+                print("\t%s\t\tRSSI: %d" % (str(access_pt['ssid'], 'utf-8'), access_pt['rssi']))
+        while not self._esp.is_connected:
+            try:
+                if self.debug:
+                    print("Connecting to AP...")
+                self.neo_status((100, 0, 0))
+                self._esp.connect_AP(bytes(self.ssid, 'utf-8'), bytes(self.password, 'utf-8'))
+                self.neo_status((0, 100, 0))
+            except (ValueError, RuntimeError) as error:
+                if self.debug:
+                    print("Failed to connect, retrying\n", error)
+                continue
```

should this failure message be printed even if debug not enabled? As it is, it just fails silently and keeps retrying.

---

Yeah, I can remove the only show on debug condition